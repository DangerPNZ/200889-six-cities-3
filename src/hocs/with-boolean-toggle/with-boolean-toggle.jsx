import React from 'react';

export const withBooleanToggle = (Component) => {
  return class WithBooleanToggle extends React.PureComponent {
    constructor(props) {
      super(props);
      this.state = {
        isActive: false
      };
      this.handleToggle = this.handleToggle.bind(this);
    }
    handleToggle() {
      this.setState((prevState) => ({isActive: !prevState.isActive}));
    }
    render() {
      return <Component
        {...this.props}
        isActive = {this.state.isActive}
        onActiveToggle = {this.handleToggle}
      />;
    }
  };
};
