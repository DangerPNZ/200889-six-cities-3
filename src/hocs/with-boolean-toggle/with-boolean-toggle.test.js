import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import {withBooleanToggle} from './with-boolean-toggle.jsx';

const TestComponent = ({isActive, onActiveToggle}) => (
  <div>
    {isActive === false && <div>isActive prop value equal to false</div>}
    <button type="button" onClick={() => onActiveToggle()}>Just a button</button>
  </div>
);
const ResultComponent = withBooleanToggle(TestComponent);

it(`Test component, returned by HOC withBooleanToggle`, () => {
  const tree = renderer
  .create(
      <ResultComponent/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

TestComponent.propTypes = {
  isActive: PropTypes.bool,

  onActiveToggle: PropTypes.func
};

