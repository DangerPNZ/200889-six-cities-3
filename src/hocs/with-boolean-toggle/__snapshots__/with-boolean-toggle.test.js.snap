// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component, returned by HOC withBooleanToggle 1`] = `
<div>
  <div>
    isActive prop value equal to false
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Just a button
  </button>
</div>
`;
